### PR TITLE
feat: block wiring of Kalix Components inside other components

### PR DESCRIPTION
**Pull request by octonato**
References  https://github.com/lightbend/kalix/issues/7367

### CODE REVIEW

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala [hunk 1]
```diff
@@ -36,4 +38,17 @@ class KalixConfiguration(applicationContext: ApplicationContext) {
...
+    override def postProcessBeforeInitialization(bean: AnyRef, beanName: String): AnyRef = {
+      if (KalixServer.kalixComponents.exists(_.isAssignableFrom(bean.getClass)))
+        throw new IllegalArgumentException(
+          s"${bean.getClass.getName} is Kalix component and is marked as a Spring bean for automatic wiring. " +
+          "Kalix components cannot be accessed directly and therefore cannot be wired into other classes. " +
+          "In order to interact with a Kalix component, you should call it using the provided KalixClient.")
+
+      bean
+    }
+  }
```
Comment (octonato) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala:52]: This is a better way of block wiring of Kalix components. In my previous commit (df7acee24699394021c53cd2aa5fd271361118d7), I was trying to simply scan the component constructor to look for param that were also Kalix components. 

That was flawed. One could by-pass the check by wrapping a Kalix component into another class, like in:

```scala
class AnotnerAction extends Action
class SomeComponent(action: AnotnerAction)
class OneAction(someComponent: SomeComponent) extends Action
```

When checking the constructor of `OneAction` we won't find `AnotnerAction`, we will only see `SomeComponent`

With a `BeanPostProcessor`, we can throw an exception when the user mark any Kalix Components with a Spring stereotype. Btw, without using sterotype, Spring can't wire them anyway.

File: sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/action/IllDefinedAction.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package kalix.springsdk.badwiring.action;
+
+import kalix.javasdk.action.Action;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IllDefinedAction extends Action {}
```
Comment (octonato) [inline sdk/spring-sdk/src/test/java/kalix/springsdk/badwiring/action/IllDefinedAction.java:23]: what makes a Kalix component ill-defined is the mix with Spring stereotypes. That make them available for wiring in the Spring context and that's exactly what we block with the `BeanPostProcessor`

**Review by efgpinto — APPROVED** [review]
Quite a nit solution after all I think. LGTM

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala [hunk 3]
```diff
@@ -36,4 +38,17 @@ class KalixConfiguration(applicationContext: ApplicationContext) {
     kalix
   }
 
+  @Component
+  class KalixComponentInjectionBlocker extends BeanPostProcessor {
+    override def postProcessBeforeInitialization(bean: AnyRef, beanName: String): AnyRef = {
+      if (KalixServer.kalixComponents.exists(_.isAssignableFrom(bean.getClass)))
+        throw new IllegalArgumentException(
+          s"${bean.getClass.getName} is Kalix component and is marked as a Spring bean for automatic wiring. " +
```
Comment (octonato) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala:46]:
```suggestion
          s"${bean.getClass.getName} is a Kalix component and is marked as a Spring bean for automatic wiring. " +
```

**Pull request merged**